presto-pinot/src/main/java/io/prestosql/pinot/PinotConnection.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.pinot;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import io.prestosql.pinot.PinotClusterInfoFetcher.TimeBoundary;
+import org.apache.pinot.common.data.Schema;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.cache.CacheLoader.asyncReloading;
+import static io.prestosql.pinot.PinotColumn.getPinotColumnsForPinotSchema;
+import static java.util.Objects.requireNonNull;
+
+public class PinotConnection
```

Comment:
This class is an unnecessary abstraction. The table-related methods and state should go in `PinotMetadata`. `PinotSplitManager` only needs `getRoutingTable` and `getTimeBoundary`, which are plain delegates for `PinotClusterInfoFetcher`, so it would just depend on that class instead.